src/libraries/System.Private.CoreLib/src/System/StringComparer.cs

```diff
@@ -62,6 +62,68 @@ public static StringComparer Create(CultureInfo culture, CompareOptions options)
             return new CultureAwareComparer(culture, options);
         }
 
+        public static bool IsWellKnownOrdinalComparer(IEqualityComparer<string?>? comparer, out bool ignoreCase)
+        {
+            if (comparer is IInternalStringEqualityComparer internalStringComparer)
+            {
+                comparer = internalStringComparer.GetUnderlyingEqualityComparer(); // unwrap if necessary
+            }
+
+            if (comparer is StringComparer stringComparer)
+            {
+                return stringComparer.IsWellKnownOrdinalComparerCore(out ignoreCase);
+            }
+            else if (comparer is GenericEqualityComparer<string>)
```

Comment:
do you use "else" here for specific reason? like better codegen? if not, I would suggest remove all else's used here.

Reply:
Shouldn't affect the codegen. I think if we take Jan's suggestion then the whole if / else thing becomes moot. Will give it a try and see what falls out.